idaes/generic_models/properties/core/eos/tests/test_enrtl_verification_NaCl_KCl.py

```diff
@@ -873,36 +904,511 @@ def test_log_gamma_salt(self, model):
+        assert model.params.Liq.tau["H2O", "Na+, Cl-"].value == 9.0234
+        assert model.params.Liq.tau["Na+, Cl-", "H2O"].value == -4.5916
+        assert model.params.Liq.tau["H2O", "K+, Cl-"].value == 8.1354
+        assert model.params.Liq.tau["K+, Cl-", "H2O"].value == -4.1341
+
+    @pytest.mark.unit
+    def test_log_gamma_NaCl(self, model):
+        # start with pure water
+        # Using 0 results in division by zero errors
+        for k in model.state[1].mole_frac_phase_comp:
+            model.state[1].mole_frac_phase_comp[k].set_value(1e-12)
+
+        # Data digitized from Fig 1 [2]
+        # Form x_KCl: log(gamma_m NaCl)
+        # Important notes, data appears to be in log_10 and on a molal basis
+        # Thus, need to convert model output to this form
+        data = {1e-12: -0.1056,
+                0.0308226637233259: -0.105016851831747,
+                0.0542281089036056: -0.104718802398099,
+                0.0776335540838852: -0.104420752964451,
+                0.099844843897824: -0.103835856101046,
+                0.123608535688006: -0.103393947923243,
+                0.147013980868285: -0.102952202881418,
+                0.170419426048565: -0.102366762231417,
+                0.193824871228845: -0.102068712797769,
+                0.217230316409124: -0.10177066336412,
+                0.242098601913171: -0.101795262910288,
+                0.257473352254809: -0.101423890651408,
+                0.277248565121413: -0.101168550217065,
+                0.299149374540103: -0.101158577170881,
+                0.322913066330285: -0.101106699929557,
+                0.345124356144224: -0.10106579358282,
+                0.367335645958163: -0.101004359292059,
+                0.499767476085357: -0.10135461223899,
+                0.56747608535688: -0.100892692714428,
+                0.589687375170819: -0.100985218003855,
+                0.613451066961001: -0.101015452538631,
+                0.63685651214128: -0.101004794321336,
+                0.66026195732156: -0.100994136104041,
+                0.68366740250184: -0.101270869103099,
+                0.707072847682119: -0.101547602102156,
+                0.72964238410596: -0.101537324535479,
+                0.751376011773363: -0.101919324732628,
+                0.768762913907285: -0.101519510086571,
+                0.788155997056659: -0.102085461424947,
+                0.811561442236939: -0.102074803207652,
+                0.834966887417219: -0.102638927423062,
+                0.858372332597498: -0.102628269205767,
+                0.881777777777778: -0.103228317323222,
+                0.905183222958057: -0.103397278616147,
+                0.928588668138337: -0.103674011615205,
+                0.970384105960265: -0.103439435672056}
+
+        for x, g in data.items():
+            m = 4  # molality of solution
+            w = 1000/18  # mol H2O per kg H2O
+            k = x*m  # mol K+
+            n = m-k  # mol Na+
```

Comment:
Can `m` and `w` be assigned outside of the for loop? It seems like only `k` and `n` need to be in the loop.